aot/src/main/java/com/dylibso/chicory/experimental/aot/AotCompiler.java

```diff
@@ -159,18 +202,147 @@ private Class<?> loadClass(byte[] classBytes) {
         }
     }
 
-    private void loadExtraClass(Map<String, byte[]> classes, byte[] bytes) {
+    private String loadExtraClass(byte[] bytes) {
         Class<?> clazz = loadClass(bytes);
-        classes.put(clazz.getName(), bytes);
+        extraClasses.put(clazz.getName(), bytes);
+        return clazz.getName();
     }
 
-    private Map<String, byte[]> compileExtraClasses() {
-        Map<String, byte[]> classes = new LinkedHashMap<>();
-        loadExtraClass(classes, createAotMethodsClass(className));
+    private void compileExtraClasses() {
+        loadExtraClass(createAotMethodsClass(className));
+
+        int totalFunctions = functionImports + module.functionSection().functionCount();
+
+        // Emit the "${className}FuncGroup_${chunk}" classes:
+        // We group the wasm functions into chunks to avoid MethodTooLargeException or
+        // ClassTooLargeException. The chunk size is dynamically adjusted based on
+        // the size of the class to be loaded and the maximum size that can be loaded
+        // without causing an exception.
+        //
+        // Example: wasm file has 1024 * 15 functions.  Then the first 12k functions will be located
+        // in "${className}FuncGroup_0" and the last 3k functions will be located in
+        // "${className}FuncGroup_1".  That is if no  MethodTooLargeException or
+        // ClassTooLargeException occur when generating those two classes.  This can happen
+        // if for example a function requires alot of class constants etc.   Then default
+        // chunk size of 12k will be halved to 6k and the first 6k functions will be located in
+        // "${className}FuncGroup_0", "${className}FuncGroup_1", and  "${className}FuncGroup_2" with
+        // each class holding up to 6k of the functions.
+        //
```

Comment:
@andreaTP added this..  Hope it helps illustrate how the chunking ocurs.

Reply:
Looks great, thanks!